sklearn/utils/_display_estimator.py

```diff
@@ -0,0 +1,311 @@
+from sklearn._config import config_context
+from contextlib import closing
+from contextlib import suppress
+from io import StringIO
+import uuid
+
+
+class _VisualBlock:
```

Comment:
Should this be public for the sake of downstream developers? Or in a later release?

Reply:
I was aiming for a later release on making this public. This PR also introduces a protocol like method called `_sk_visual_block_` that defines and returns the visual blocks. (It was named `_sk_repr_html` before)

For the first release, let's see how users feel about the visualization and we can adjust as needed. Then for the next release, we can make `VisualBlock` public and allow downstream developers to use it as well.